### ISSUE TITLE
Problems when updating extensions

### ISSUE THREAD

**Issue by eggplanting** [NeverDecaf/chromium-web-store#16]
Hello,

when updating extensions, I faced two issues with this extension:

1. "Update All Extensions" does not work
When updates for multiple extensions were available at my systems, I chose "Update All Extensions". It started to download all updates for each extension, but I got only one popup which notified me to update one extension only. I didn't get popup for the other extensions. The download for the other extension was stuck. I was able to recreate this problem on multiple Windows desktops with Ungoogled Chromium (81.0.4044.113)

2. Notification for new updates doesn't get removed after updating until browser restart
When I was notified about an extension update (ublock Origin), I installed the update. The notification still was shown. This also happened when multiple update were available. In both situations, the counter stayed at "1 new update". Restarting the browser fixes this issue, until a new update is available.

Great addon, I'd be happy if you can fix those issues.

**Comment by NeverDecaf** [comment]
Unfortunately, your second issue cannot be fixed due to limitations in the extension api, see https://github.com/NeverDecaf/chromium-web-store/issues/4#issuecomment-559064509. What is essentially happening is the extensions are not actually updated until you restart the browser, hence the notification not disappearing.

The first issue is a little more complicated. Updates are currently done by opening a new tab using window.open(), with the destination being the url of the crx file. This, in combination with the `#extension-mime-request-handling` flag makes installing extensions simple, with minimal clicking involved. However, to install multiple at once using this method means opening multiple tabs at once with window.open. This is forbidden according to [this answer](https://stackoverflow.com/questions/16749907/window-open-behaviour-in-chrome-tabs-windows) and as far as I can tell the `update all` feature should not work **at all**, yet it somehow does, but *only* when run from background.js, and apparently only on certain setups (not yours).

The fix then seems obvious, just use a different method such as [chrome.downloads](https://developer.chrome.com/extensions/downloads#method-download) to download the crx files instead. This would almost certainly fix your issue, *however*, if the files are downloaded by any method besides window.open(), chromium will prompt you with `Extensions, apps, and themes can harm your computer. Are you sure you want to continue?` for each and every extension you download. This would make `update all` nearly pointless as you could manually click each extension one at a time with basically the same amount of effort as confirming this prompt for each extension.

The tl;dr is: I don't currently know of a way to fix the problem. I explain all this because I am hoping that someone out there can provide a workaround that I can implement. I will leave this issue open in case anyone has an idea to fix this.

I am also curious how many others are experiencing problem 1, please leave a comment (with OS/Chromium version) if you are also unable to `Update All Extensions` beyond the first one.

**Comment by eggplanting** [comment]
Thanks a lot for your explanation. For now, I'll update each extensions seperately and restart the browser. This is a usable workaround for me.

I'm also interested to see what experiences other users are getting, I used following OS / Chromium combinations:
- Windows 10 (1909) with Ungoogled Chromium 80.0.3987.132 (x64) and faced the issues on 3 platforms running this setup

Keep up the great work, it makes Ungoogled Chromium finally usable for me and convinced me to use it as my main browser!

**Comment by NeverDecaf** [comment]
Closing as there probably isn't a solution to this. I will note that I tried using chrome.tabs.create() in place of window.open() but doing so always creates the `Extensions, apps, and themes can harm your computer. Are you sure you want to continue?` prompt despite the headers being identical (this may have something to do with the sec-fetch-user header, will experiment more in the future).

**Comment by Tank-Missile** [comment]
I did some testing, and issue #1 works perfectly fine for me when updating multiple webstore extensions. In previous versions it was not working at all for me, so I'm not sure what changed. As for issue #2, I only encounter it when updating non-webstore extensions such as this extension. My ungoogled-chromium version is 83.0.4103.116.
